optimizer/plan/planbuilder.go

```diff
@@ -34,11 +36,7 @@ const (
 func BuildPlan(node ast.Node) (Plan, error) {
 	var builder planBuilder
 	p := builder.build(node)
-	if builder.err != nil {
-		return nil, builder.err
-	}
-	err := refine(p)
-	return p, err
+	return p, builder.err
```

Comment:
The function only have one place to return this error, we don't need to trace it.